hw3/01-routing.js

```diff
@@ -24,26 +25,63 @@ const server = http.createServer((req, res) => {
     'other',
   ];
 
-  let getRoutes = () => {
+  const getRoutes = () => {
     let result = '';
 
     routes.forEach(
-      (elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`)
+      (elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`),
     );
 
     return result;
   };
 
   if (req.url === '/') {
-    let routeResults = getRoutes();
+    const routeResults = getRoutes();
 
     res.writeHead(200, { 'Content-Type': 'text/html' });
-    res.write(`<h1>Exercise 01</h1>`);
+    res.write('<h1>Exercise 01</h1>');
     res.write(`<ul> ${routeResults} </ul>`);
     res.end();
+  } else if (req.url === '/welcome') { // Route for welcome page
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write('<h1>Welcome to Homework 3 exercise 1</h1>');
+    res.end();
+  } else if (req.url === '/redirect') { // Route for redirect
+    res.writeHead(302, { location: '/redirected' });
+    res.end();
+  } else if (req.url === '/redirected') { // Route for redirected page
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write('<h1>You have been redirected</h1>');
+    res.end();
+  } else if (req.url === '/cache') { // Route for cache page
+    res.writeHead(200, {
+      'Content-Type': 'text/plain',
+      'Cache-Control': 'max-age=<86400>',
+    });
+    res.write('This resource was cached');
+    res.end();
+  } else if (req.url === '/cookie') { // Route for cookie page, sets a cookie
+    res.writeHead(200, {
+      'Content-Type': 'text/plain',
+      'Set-Cookie': 'hello=world',
+    });
+    res.write('cookies... yummmm');
+    res.end();
+  } else if (req.url === '/check-cookies') { // Route for check-cookies
+    let print = false;
+    res.writeHead(200, { 'Content-Type': 'text/plain' });
+    let cookies = req.headers.cookie;
+    cookies = cookies.split(';');
+    cookies.forEach((i) => {
+      if (i === ' hello=world') { print = true; }
```

Comment:
I went about it this same way too but I'm wondering if there is a different way to compare just the hello by itself. I remember her briefly mentioning an example if there were multiple cookies that were to start with the same word but have something different after the = (or vice versa). I just couldn't get it to work myself, so my code is the same too. I'm just curious though if we were allowed to do it this way lol.